src/UI/Component/Dropzone/File/Dropzone.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Interface Dropzone
+ *
+ * Describes a dropzone that can be configured to use the darkened background
+ * or not.
```

Comment:
> For me it seems to be better to start to think about the (abstract) base or standard interface first. Put everything in it that has not a clear reference to files or dependencies of files

Seems reasonable.

> And I think we might not need different ways of highlighting/focusing drop zones.`

We do need the one with the darkened bg. See FR here: http://www.ilias.de/docu/goto.php?target=wiki_1357_File_Upload_for_all_Appointment_Types The other one (without darkened background is thought to be useful e.g. in forms. However, this is open for discussion. We can also start with just the darkened bg version and add the other one later if the need arises. Personally I would rather have a good concept for both now. To me it seems like we will need both soon. However I do not have a strong opinion on this as long as we keep the one with the backdrop.

> The darkened background uses the same bootstrap classes like the modals. So there is no less variable for it.

Actually there are: modal-backdrop-bg and modal-backdrop-opacity. Just realized, that there is some semantic bump. You add the class modal-backdrop per JS. Semantically this makes no sense, since we do not really have a modal on the screen at this point. I would rather recommend to extend this modal class e.g.:

```
.dropzone-backdrop {
  &:extend(.modal-backdrop);
  background-color: @dropzone-backdrop-bg;
  &.in { .opacity(@dropzone-backdrop-opacity); }
}
```

and in the variables less of ILIAS in the dropzone section you do something like:

```
//**Sets the background color of a focused dropzone with activated backdrop
@dropzone-backdrop-bg: @modal-backdrop-bg;
//**Sets the opacity of a focused dropzone with activated backdrop
@dropzone-backdrop-opacity: @modal-backdrop-opacity;
```